app/models/manageiq/providers/ibm_power_hmc/inventory/parser/infra_manager.rb

```diff
@@ -34,7 +34,37 @@ def parse_ssps
         :name        => ssp.name,
         :total_space => ssp.capacity.to_f.gigabytes.round, # hmc returns a str in byte
         :ems_ref     => ssp.uuid,
-        :free_space  => ssp.free_space.to_f.gigabytes.round
+        :free_space  => ssp.free_space.to_f.gigabytes.round 
+      )
+    end
+  end
+
+  def parse_vm_disks(lpar, hardware)
+    lpar_lu_mappings = collector.vioses.flat_map do |vios|
+      vios.vscsi_mappings.select do |mapping|
+        mapping.lpar_uuid == lpar.uuid && mapping.storage.kind_of?(IbmPowerHmc::LogicalUnit)
+      end
+    end
+
+    lpar_lu_mappings.each do |mapping|
+      found_ssp = nil
+      collector.ssps.each do |ssp|
+        lu = ssp.lus.find do |lu|
+          lu.udid == mapping.storage.udid
+        end
+        unless lu.nil?
+          found_ssp = ssp
+          break
+        end
+      end
+
+      $ibm_power_hmc_log.info("#{self.class}##{__method__} : found_ssp : #{found_ssp}")
```

Comment:
Is the `found_ssp` used for anything other than logging?

Reply:
I wanted to link `found_ssp`  as the `:storage` attribute in `disks` table. 
Something like : 
```
persister.disks.build(
    :hardware    => hardware,
    :device_type => "disk",
    :storage     => found_ssp,
    :device_name => mapping.storage.name,
    :size        => mapping.storage.capacity.to_f.gigabytes.round
)
```

But I think it changes nothing in my table disks.

Reply:
Ah okay, try `persister.storages.lazy_find(found_ssp)` (this assumes that found_ssp matches the manager_ref of your storages collection)